### PR TITLE
feat: Deselect items after moving to window or folder. 

### LINKED ISSUE #228: Option to deselect marked items once moved to another stash group
### Problem Statement
If multiple tabs entries are selected (blue icon with tab symbol) and moved to another group using the blue command button at the top, the items all remain selected inside their target group. If that target group was collapsed (or at the very bottom of the list) it's tedious to scroll there and open the target group and un-mark all items again.

### Preferred Solution(s)
Remove selected status once the items have been moved. 

I can't think of any follow up action that would required them to remain selected.

### Alternative Solution(s)
Provide an option in the "blue" command menu to deselect these items.

The only option in that menu is "Delete and close". I have no idea what that's supposed to mean or what it refers to. It sounds not like something I want to do with the items I have just decided to move someplace else.

### Additional Context

<hr>
<em>Readers: If you are also interested in seeing this feature be developed, please vote for it by giving the ORIGINAL POST a thumbs-up using the :smiley: button.  You are welcome to leave comments and discuss the feature request, but "Me too!" comments are not counted by the voting system.</em>

### PR DESCRIPTION
**Pull request by KerfuffleV2**
Does not apply when copying them, since I'd assume people might want to perform more actions with the selected items.

Note: Right now this is implemented by creating a general `putSelectedIn` function and the existing `putSelectedInWindow` and `putSelectedInFolder` functions are just wrappers.

I think there's an argument for leaving it like this: It makes it more clear what the action is actually doing, and it would allow adding custom behavior for folders vs windows at some point in the future.  It would be very easy to remove the wrappers and change the templates to just call `putSelectedIn` directly, so please let me know if you'd prefer that approach.

Closes #228

### CODE REVIEW

**Comment by KerfuffleV2** [comment]
Also a question — what's up with the semicolon in the type here?
```typescript
async putSelectedInWindow(options: {copy: boolean;}) { /* ... */ }
```

Does it actually do something? I haven't seen that before.

**Comment by josh-berry** [comment]
> Also a question — what's up with the semicolon in the type here?
> 
> ```ts
> async putSelectedInWindow(options: {copy: boolean;}) { /* ... */ }
> ```
> 
> Does it actually do something? I haven't seen that before.

Nope, I was probably just being pedantic about semicolons.

File: src/model/index.ts [hunk 1]
```diff
@@ -289,19 +289,37 @@ export class Model {
...
+    async putSelectedIn(
+        options?: {
+            copy?: boolean, toFolderId?: Bookmarks.NodeID,
+        }) {
+        const from_items = Array.from(this.selectedItems());
+        const items = copyIf(options?.copy === true, from_items);
+
+        if (!options?.copy) {
+            from_items.forEach(i => i.$selected = false);
+        }
```
Comment (josh-berry) [inline src/model/index.ts:305]: I think I'd wait to do this until after `putItemsIn...()` for two reasons. First, if `putItemsIn...()` throws for some reason, the original selection will be left intact. And second, it looks a little less confusing--sometimes there's a noticeable delay when moving a large number of items. Leaving the original items selected will show the items moving along with their selection status, so it's a little easier to tell where the items are being placed.

**Review by josh-berry — COMMENTED** [review]
I think the refactoring is good; with your change, there's enough common logic that it makes sense to push it down to a common function.

File: src/model/index.ts (hunk 1 above)
Comment (KerfuffleV2) [reply src/model/index.ts]: Makes sense!

**Pull request merged**